### PR TITLE
[virt-api]: Add Prometheus records & alerts for bad REST calls

### PR DESCRIPTION
**Pull request by iholder-redhat**
**What this PR does / why we need it**:
This PR adds Prometheus records to track total amount of REST calls in last hour & 5 minutes, and bad REST calls over the same time-frame:
- `vec_by_virt_apis_all_client_rest_requests_in_last_5m`
- `vec_by_virt_apis_all_client_rest_requests_in_last_hour`
- `vec_by_virt_apis_failed_client_rest_requests_in_last_5m`
- `vec_by_virt_apis_failed_client_rest_requests_in_last_hour`

These records then set these alert rules:
- `VirtApiRESTErrorsHigh`: `More than 5% of the rest calls failed in virt-api for the last hour`
- `VirtApiRESTErrorsBurst`: `More than 80% of the rest calls failed in virt-api for the last 5 minutes`

**Special notes for your reviewer**:
I'm not entirely sure where `prom-rules-tests.yaml` is used besides testing. Does it have any non-testing usage? How exactly does it help for testing? How was it generated?

**Release note**:

```release-note
Fire a Prometheus alert when a lot of REST failures are detected in virt-api
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -160,14 +160,19 @@ olm-verify:
...
+
+clean-prom-spec-dumper:
+	rm -f ${rule-spec-dumper-executable}
 
 prom-rules-verify: build-prom-spec-dumper
 	./hack/prom-rule-ci/verify-rules.sh \
-		"${current-dir}/rule-spec-dumper" \
+		"${current-dir}/${rule-spec-dumper-executable}" \
 		"${current-dir}/hack/prom-rule-ci/prom-rules-tests.yaml"
+	$(MAKE) clean-prom-spec-dumper
```
Comment (dhiller) [inline Makefile:175]: Being curious: why is the rebuild required every time?

**Review by dhiller — COMMENTED** [review]
/approve

@sradco mind taking a look here? Thanks!

File: Makefile (hunk 1 above)
Comment (iholder-redhat) [reply Makefile]: Thanks Daniel. I've removed this.

**Comment by sradco** [comment]
@iholder-redhat The alerts are missing runbooks here https://github.com/kubevirt/monitoring/tree/main/docs/runbooks 
and runbook_url annotation
See example:
https://github.com/kubevirt/kubevirt/blob/42a49ae61d3b41a78439acfcf1951507dfa61915/pkg/virt-operator/resource/generate/components/prometheus.go#L102
https://github.com/kubevirt/monitoring/blob/main/docs/runbooks/VirtAPIDown.md

**Comment by iholder-redhat** [comment]
> @iholder-redhat The alerts are missing runbooks here https://github.com/kubevirt/monitoring/tree/main/docs/runbooks and runbook_url annotation See example:
> 
> https://github.com/kubevirt/kubevirt/blob/42a49ae61d3b41a78439acfcf1951507dfa61915/pkg/virt-operator/resource/generate/components/prometheus.go#L102
> 
> 
> https://github.com/kubevirt/monitoring/blob/main/docs/runbooks/VirtAPIDown.md

No problem, I'll add it up

**Pull request merged**